### ISSUE TITLE
Many fixed-output derivations fail with `...chroot/etc/nsswitch.conf: Permission denied`

### ISSUE THREAD

**Issue by zseri** [NixOS/nix#7866]
**Describe the bug**

While trying to update NixOS, I hit the following problem with multiple fixed-output derivations:
```
error:
       … while setting up the build environment

       error: opening file '/nix/store/7as84y1b7r5vmrgzzsihzsw58ai6mk26-txtorcon-23.0.0.tar.gz.drv.chroot/etc/nsswitch.conf': Permission denied
```

**Steps To Reproduce**

unknown

**Expected behavior**

the build should succeed. setting up the build environment should be reliable.

**`nix-env --version` output**

nix-env (Nix) 2.14.0pre20230215_601849b

**Additional context**

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by SuperSandro2000** [comment]
Should be fixed by #7856 and be a dupe of #7852

**Comment by zseri** [comment]
thx
